src/chat/backend.rs

```diff
@@ -105,6 +113,32 @@ impl CallCodec {
             sender: sender,
         }
     }
+
+    fn add_request_id<S>(&self, e: &mut http::Encoder<S>) {
+        if let Some(ref header) = self.destination.request_id_header {
+            let rid = match self.meta.get("request_id") {
+                Some(&Json::String(ref rid)) if rid.len() <= 36 => {
+                    if rid.chars().all(|c| {
+                        c.is_digit(36) || c == '-' || c == '_' })
+                    {
+                        format!("{}-{}-{}", self.server_id, self.conn_id, rid)
+                    } else {
+                        format!("{}-{}-{}", self.server_id, self.conn_id,
+                            request_id::new())
+                    }
+                }
+                Some(&Json::Number(ref n)) if n.is_u64() => {
+                    format!("{}-{}-{}", self.server_id, self.conn_id,
+                        n.as_u64().unwrap())
+                }
+                _ => {
```

Comment:
Reject closing socket?
The only valid `request_id` would be either `u64` or alphanumeric strings up to 36 chars.